homeassistant/components/upnp/config_flow.py

```diff
@@ -134,6 +134,15 @@ async def async_step_ssdp(self, discovery_info: Mapping):
         """
         _LOGGER.debug("async_step_ssdp: discovery_info: %s", discovery_info)
 
+        # Ensure complete discovery.
+        if (
+            ssdp.ATTR_UPNP_UDN not in discovery_info
+            or ssdp.ATTR_SSDP_ST not in discovery_info
+            or "friendlyName" not in discovery_info
```

Comment:
The friendly name is optional? See line 154:

`name = discovery_info.get("friendlyName", "")`

It defaults to an empty string. This will change that behavior. Could you elaborate on that?

Reply:
Seems to be an mistake, removed.